src/@types/events.ts

```diff
@@ -3,8 +3,9 @@ enum Event {
     ACTIVE_SET = 'annotations_active_set',
     ANNOTATION_CREATE = 'annotations_create',
     ANNOTATION_FETCH_ERROR = 'annotations_fetch_error',
-    ANNOTATIONS_INITIALIZED = 'annotations_initialized',
     ANNOTATION_REMOVE = 'annotations_remove',
+    ANNOTATIONS_INITIALIZED = 'annotations_initialized',
+    ANNOTATIONS_MODE_CHANGE = 'annotations_mode_change',
```

Comment:
Should this be `annotations_mode_toggle` to match the public-facing method and action name?

Reply:
I kind of like `annotations_mode_change` since the event returns the change, whereas the method and action are actually toggling a given mode, i.e. `{ mode: 'region' }`, `toggeMode('region')` emits `annotation_mode_change('none')`

Reply:
It seems like the logical end result with that approach will require two events for each state change, one in and one out. It seems like you get the same result with fewer events by either emitting into or listening for the same event on the annotator instance. What do you think?

Reply:
I like `annotations_mode_change` as well since it is just informing the listener that the mode changed and it can toggle or do what ever it needs, I think it gives more control.